crates/swc_ecma_minifier/src/compress/pure/strings.rs

```diff
@@ -121,13 +121,15 @@ impl Pure<'_> {
             quasis: Default::default(),
             exprs: Default::default(),
         };
-        let mut cur_str_value = String::new();
+        let mut cur_cooked_str = String::new();
+        let mut cur_raw_str = String::new();
 
         for idx in 0..(tpl.quasis.len() + tpl.exprs.len()) {
             if idx % 2 == 0 {
                 let q = tpl.quasis[idx / 2].take();
 
-                cur_str_value.push_str(q.cooked.as_deref().unwrap_or(&*q.raw));
+                cur_cooked_str.push_str(q.cooked.as_deref().unwrap_or(&*q.raw));
```

Comment:
I think this logic is wrong. `cooked` or `raw` should never be the case, and we should run tpl string replace logic for `raw` if we want to use it as a `value`.

See https://rustdoc.swc.rs/swc_ecma_ast/struct.Str.html#method.from_tpl_raw

Reply:
updated